### DISCUSSION TITLE
Can we handle finite/small faults (that do not go across through geomodel extents)  in GemPy?

### DISCUSSION THREAD

**Discussion by geomodeller** [gempy-project/gempy#675] [Q&A] [answered]
Hi,

I wish to ask if there is a way to include finite faults in GemPy. For example, the following figure shows a top view of geomodel with three faults. 

![image](https://user-images.githubusercontent.com/46004761/155377305-2f179455-23e0-4c1b-898f-39c41bc5cd3b.png)

I know how to include faults like "fault #1" that go across the extent. However, "fault #2 and #3" does not go up to the geomodel extent, but when importing in GemPy, I've seen that it will create faults like:

![image](https://user-images.githubusercontent.com/46004761/155377827-bb603775-678f-48ee-b030-e483a3a59b87.png)

In a way to make them finite faults, I was thinking of adding two imaginary faults (i.e., green) that can define the ends of individual faults, as shown below:
 
![image](https://user-images.githubusercontent.com/46004761/155378218-a07323f0-c3f3-4fec-bda5-fb5402ef8fb9.png)

To make those green faults not affect the surfaces, we set the faults to have "isFinite == False". But it didn't work as we wished :(

Any help would be highly appreciated!

Thank you,
Best!

Honggeun Jo

**Comment by geomodeller** [comment]
Oh, we set green faults' "isFinite" == True, not False

**Comment by Japhiolite** [comment, marked as answer]
Hi @whghdrms 
Yes, indeed you can set faults as finite with the "isFinite" method. Be cautious, though, that method needs some work, don't trust the results too much.
Also, the faults will still be plotted across the whole model domain, but that is just visualization. Their influence on the model is more or less confined to the extent of the input-points for the respective fault.

**Reply by geomodeller** [reply]
Thank you for your prompt response! After posting this question, I have been 'playing' with GemPy to have two perpendicular faults where one of the faults goes across the other one while not affecting strata. Here is what I got:

![image](https://user-images.githubusercontent.com/46004761/155463461-260ba05e-e5f1-4e0b-a040-04162c639cef.png)

![image](https://user-images.githubusercontent.com/46004761/155463403-d54a9c5c-e4a4-4550-b6f9-a9c3764f4911.png)


On top of setting "isFinite == True," we also need to (1) sequence of fault #2 (i.e., an imaginary fault that define the boundary) comes after fault #1 and (2) change the relationship of fault as shown below:
![image](https://user-images.githubusercontent.com/46004761/155463720-730144bb-228c-4cdd-bed7-fa1d760830c8.png)

I also found some visual artifacts, but when transformed to NumPy.array, the result looks reasonable. Thank you again for your help!

Best,
Honggeun Jo

**Reply by Japhiolite** [reply]
Good looking model Honggeun, nicely presents fault relationships. The visual artifacts may occur yes, due to the main_fault stopping at main_fault_2 and being deflected along the surface of main_fault_2 using a marching cubes algorithm.
